src/crimson/os/seastore/transaction.h

```diff
@@ -329,7 +329,7 @@ class Transaction {
     auto retired_paddr = (*iter)->get_paddr();
     auto retired_length = (*iter)->get_length();
     return retired_paddr <= paddr &&
-      retired_paddr.add_offset(retired_length) >= paddr.add_offset(len);
```

Comment:
Follow up https://github.com/ceph/ceph/pull/57709#discussion_r1643641271, does it look more accurate as a simple accurate check?

```c++
bool is_retired(paddr_t paddr, extent_len_t len) {
  auto iter = retired_set.lower_bound(paddr);
  if (iter == retired_set.end() ||
      (*iter)->get_paddr() != paddr) {
    return false;
  } else {
    assert(len == (*iter)->get_length());
    return true;
  }
}
```

Can you merge the 1st and the last commits for the same purpose?

Reply:
Ah, yes, will do

Reply:
Done:-)